tests/test_converter.py

```diff
@@ -1397,6 +1399,104 @@ def test_diff_meas_circuit(self):
         qtemp2 = load(qc, measurements=[qml.expval(qml.PauliZ(0))])
         assert qtemp()[0] != qtemp2()[0] and qtemp2()[0] == qml.expval(qml.PauliZ(0))
 
+    def test_control_flow_ops_circuit_ifelse(self):
+        """Tests mid-measurements are recognized and returned correctly."""
+
+        qc = QuantumCircuit(3, 3)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.measure(0, 0)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.measure(0, 1)
+
+        with qc.if_test((0, 0)) as else_:
+            qc.x(0)
+
+        with else_:
+            qc.h(0)
+            qc.z(2)
+
+        qc.rz(0.24, [0])
+        qc.cx(0, 1)
+        qc.measure_all()
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def loaded_qiskit_circuit():
+            meas = load(qc)()
+            return [qml.expval(m) for m in meas]
+
+        @qml.qnode(dev)
+        def built_pl_circuit():
+
+            qml.Hadamard(0)
+            qml.CNOT([0, 1])
+            m0 = qml.measure(0)
+            qml.Hadamard(0)
+            qml.CNOT([0, 1])
+            m1 = qml.measure(0)
+
+            def ansatz_true():
+                qml.Hadamard(wires=0)
+                qml.PauliZ(wires=2)
+
+            def ansatz_false():
+                qml.PauliX(wires=0)
+
+            qml.cond(m0, ansatz_true, ansatz_false)()
+
+            qml.RZ(0.24, wires=0)
+            qml.CNOT([0, 1])
+
+            return [qml.expval(m) for m in [m0, m1, qml.measure(0), qml.measure(1), qml.measure(2)]]
+
+        assert loaded_qiskit_circuit() == built_pl_circuit()
```

Comment:
```suggestion
        assert loaded_qiskit_circuit() == built_pl_circuit()
        assert qml.equal(loaded_qiskit_circuit.tape, built_pl_circuit.tape)
```
Might also be worth checking this for each of the tests.

Reply:
Tried doing it, and the operations comes out to be same but the `qml.equal` returns `False` because `qml.measure` assertion fails 🤔 

```
assert [Hadamard(wires=[0]), CNOT(wires=[0, 1]), measure(wires=[0]), Hadamard(wires=[0]), CNOT(wires=[0, 1]),
 measure(wires=[0]), Conditional(wires=[0]), Conditional(wires=[0]), Conditional(wires=[2]), RZ(0.24, wires=[0]),
 CNOT(wires=[0, 1]), measure(wires=[0]), measure(wires=[1]), measure(wires=[2])] == 
[Hadamard(wires=[0]), CNOT(wires=[0, 1]), measure(wires=[0]), Hadamard(wires=[0]), CNOT(wires=[0, 1]), 
 measure(wires=[0]), Conditional(wires=[0]), Conditional(wires=[0]), Conditional(wires=[2]), RZ(0.24, wires=[0]),
 CNOT(wires=[0, 1]), measure(wires=[0]), measure(wires=[1]), measure(wires=[2])]

E         At index 2 diff: measure(wires=[0]) != measure(wires=[0])
```

Reply:
Added a manual check for testing it.